### ISSUE TITLE
Should abstract length be limited

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#2536]
Today I encountered the following error.  I grant that the abstract I was intending to enter was long - but it was the award abstract from https://nsf.gov/awardsearch/showAward?AWD_ID=1664061 in a resource where I was sharing the HydroShare proposal.  I would prefer that length of items not be arbitrarily limited wherever possible.

![image](https://user-images.githubusercontent.com/3456833/34263760-ce2e844a-e62d-11e7-8666-ec687dcebd3e.png)

**Comment by cuahsimarko** [comment]
@zhiyuli I think this is related to the problems you're encountering for the project that's importing production data to HydroShare for CZO can you confirm?

**Comment by cuahsimarko** [comment]
This is an edge case for CZO project, < 10 instances. Will rely on manual remediation in comments and/or markdown files.

**Comment by cuahsimarko** [comment]
Duplicate/related issue: https://github.com/hydroshare/hydroshare/issues/3125

**Comment by horsburgh** [comment]
I don't think that an abstract length limitation to 5000 characters is unreasonable. Users can upload content files to the resource, and they can write whatever they want in a readme file. The Abstract doesn't need to be a book. However, if there is a length restriction, it should be indicated on the page in edit mode so users can clearly see it when they are editing the resource.

@dtarb - I'm inclined to just add a label to the edit page with the length restriction on the Abstract field, but if you feel strongly that the Abstract should not be limited in length, then we would have to change that metadata field.

**Comment by dtarb** [comment]
The instance that triggered this was a real NSF project summary.  We should avoid arbitrary limiting restrictions.  If it is technically possible to have a field type with flexible or unlimited length, I would prefer that.  If this is not technically feasible, I am OK with imposing a maximum length and indicating this on the edit page.  I suggest the limit be longer than 5000 characters.  I suggest 10000 (but really any number we set is arbitrary).

**Comment by sblack-usu** [comment]
It is not difficult to increase the abstract length and currently the only limitation is in the form, not the database.  If we remove the limit from the form, no other changes should be necessary.  Relevant code below.
https://github.com/hydroshare/hydroshare/blob/develop/hs_core/models.py#L679
https://github.com/hydroshare/hydroshare/blob/develop/hs_core/forms.py#L810

**Comment by horsburgh** [comment]
@alvacouch - can you confirm whether there are any implications for very long abstract text on metadata cataloging and discovery performance? If not, I guess we can make this change.

I still don't think it's necessary, though. The purpose of an Abstract is to provide a complete, but **concise** statement about the content of the resource. Many journals limit this to 150 -250 words. I think this encourages overloading the Abstract with a bunch of stuff that doesn't belong there.
